cupyx/fallback_mode/fallback.py

```diff
@@ -17,10 +17,21 @@ class _RecursiveAttr:
     when user calls fallback_mode. numpy is an instance of this class.
     """
 
-    def __init__(self, numpy_object, cupy_object):
+    def __init__(self, numpy_object, cupy_object, array=None):
 
         self._numpy_object = numpy_object
         self._cupy_object = cupy_object
+        self._fallback_array = array
+
+    def __instancecheck__(self, instance):
+        """
+        Enable support for isinstance(instance, _RecursiveAttr instance)
+        by redirecting it to appropriate isinstance method.
+        """
+        if self._cupy_object is not None:
+            return isinstance(instance, self._cupy_object)
```

Comment:
Bug. The test seems not sufficient.
```py
>>> import cupy
>>> from cupyx.fallback_mode import numpy
>>> isinstance(numpy.ndarray([1]), object)
True
```

Reply:
It's not a bug. All classes in Python3 are new-style classes. They are always derived from `object`. `__instancecheck__` is only redirecting `isinstance()`. `numpy` and `cupy` also shows same behavior.
```python
>>> import cupy
>>> isinstance(cupy.ndarray([1]), object)
True
>>> import numpy
>>> isinstance(numpy.ndarray([1]), object)
True
>>>
```

Reply:
Sorry, it's my terrible typo. I wanted to say the following thing.
```
>>> from cupyx.fallback_mode import numpy
>>> import cupy
>>> x = numpy.array([1])
>>> isinstance(x, cupy.ndarray)
True
```

Reply:
It is related to https://github.com/cupy/cupy/pull/2272#discussion_r300536549.

Reply:
We can continue here: https://github.com/cupy/cupy/pull/2272#discussion_r300535716